### ISSUE TITLE
Ordering issues with php::global applying before PHP packages installed

### ISSUE THREAD

**Issue by chrisboulton** [mayflower/puppet-php#191]
I've come across a strange ordering issue, where `php::global` seems to be included and attempting to apply resources (`Php::Global/Php::Config[global]/Php::Config::Setting[/etc/php5/php.ini: PHP/variables_order]`) before PHP is actually installed - at this point in time `/etc/php5/` doesn't exist, which causes a bunch of errors:

```
==> default: Error: Could not set 'present' on ensure: No such file or directory - /etc/php5/php.ini at 54:/var/tmp/vagrant-puppet/modules-c61fb460434b0ef3bc9fb916e18be6c3/php/manifests/config/setting.pp
==> default: Error: Could not set 'present' on ensure: No such file or directory - /etc/php5/php.ini at 54:/var/tmp/vagrant-puppet/modules-c61fb460434b0ef3bc9fb916e18be6c3/php/manifests/config/setting.pp
==> default: Wrapped exception:
==> default: No such file or directory - /etc/php5/php.ini
```

Down later in the Puppet run, the PHP packages are installed, and the directory structure is good. A second run succeeds.

A quick glance into init.pp seems to have this handled with the anchoring, so at this point in time I'm not sure why this is happening. I haven't dug further yet.

**Comment by chrisboulton** [comment]
Actually, is the anchoring right?

```
  # Configure global PHP settings in php.ini
  Anchor['php::begin'] ->
  class {'::php::global':
    settings => $real_settings,
  } ->
  Anchor['php::end']
```

I'd have expected it to be the other way around - `php::end` must be complete before the settings (aaaand FPM down lower?) can be applied:

```
  # Configure global PHP settings in php.ini
  class {'::php::global':
    settings => $real_settings,
    require  => Anchor['php::end'],
  }
```

**Comment by mmoll** [comment]
should be fixed by #194
